### PR TITLE
case 21374 : Away mode on Quest when device is paused

### PR DESCRIPTION
**Pull request by amerhifi**
https://highfidelity.manuscript.com/f/cases/21374/

Setting away mode when user enters pause state on the device by either taking it off their head or hitting the oculus button.

Test:

1. Enter domain with PC or other device
2. Enter the same domain as the device above using the quest
3.  Make sure the avatars can see each other
4. Press Oculus button on the quest to enter pause mode
5. On the pc, observe the avatar go into kneel animation
6. On Quest hit resume.
5. Observe the avatar (using the other device) go back to active (non away mode)

### CODE REVIEW

File: android/libraries/oculus/src/main/java/io/highfidelity/oculus/OculusMobileActivity.java [hunk 1]
```diff
@@ -123,8 +128,7 @@ public void surfaceChanged(SurfaceHolder holder, int format, int width, int heig
     @Override
     public void surfaceDestroyed(SurfaceHolder holder) {
         Log.w(TAG, "QQQ_ surfaceDestroyed ***************************************************");
-      //  nativeOnSurfaceChanged(null);
-       // mSurfaceHolder = null;
-
```
Comment (amerhifi) [inline android/libraries/oculus/src/main/java/io/highfidelity/oculus/OculusMobileActivity.java]: this was a left over from last pr that I forgot to re-enable

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -9158,6 +9163,15 @@ void Application::enterForeground() {
     auto nodeList = DependencyManager::get<NodeList>();
     nodeList->setSendDomainServerCheckInEnabled(true);
 }
+
+
+void Application::toggleAwayMode(){
+   QKeyEvent event = QKeyEvent (QEvent::KeyPress, Qt::Key_Escape, Qt::NoModifier);
+    QCoreApplication::sendEvent (this, &event);
```
Comment (amerhifi) [inline interface/src/Application.cpp]: I wanted to leverage the away mode animation from the script with minimal code duplication. Firing escape key event seemed to be the easiest way to replicate that.

File: scripts/+android_questInterface/defaultScripts.js [hunk 3]
```diff
@@ -14,8 +14,8 @@
 var DEFAULT_SCRIPTS_COMBINED = [
     "system/request-service.js",
     "system/progress.js",
-    //"system/away.js",
-    "system/hmd.js",
+    "system/away.js",
+    //"system/hmd.js",
```
Comment (amerhifi) [inline scripts/+android_questInterface/defaultScripts.js]: Disabled hmd script to remove "exit vr mode"

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17512/HighFidelity-Beta-PR14986-9c833f7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17512/HighFidelity-Beta-PR14986-9c833f7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17512/HighFidelity-Beta-Interface-PR14986-9c833f7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17512/HighFidelity-Beta-PR14986-9c833f7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17512/HighFidelity-Beta-Interface-PR14986-9c833f7.exe

**Pull request merged**